src/cpprinterface.cpp

```diff
@@ -0,0 +1,26 @@
+#include "cpp11.hpp"
+
+#include <R.h>
+#include <Rinternals.h>
+#include <Rdefines.h>
+#include <R_ext/Visibility.h>
+
+#include "igraph_vector.hpp"
+
+extern "C" int igraphhcass2(int *n, int *ia, int *ib, int *iorder, int *iia, int *iib);
+
+[[cpp11::register]]
+SEXP igraph_hcass2(SEXP n, SEXP ia, SEXP ib) {
```

Comment:
Let's use cpp11 to help us here.

```suggestion
SEXP igraph_hcass2(int c_n, SEXP ia, SEXP ib) {
```

Ideally, we could also do

```c
SEXP igraph_hcass2(int c_n, igVector a, igVector b) {
```

to harness the full power.